Gems/Atom/RPI/Code/Source/RPI.Reflect/Image/StreamingImageAsset.cpp

```diff
@@ -93,18 +93,50 @@ namespace AZ
         {
             return m_totalImageDataSize;
         }
-        
+
+        RHI::ImageDescriptor StreamingImageAsset::GetImageDescriptorForMipLevel(AZ::u32 mipLevel) const
+        {
+            RHI::ImageDescriptor imageDescriptor = GetImageDescriptor();
+
+            const ImageMipChainAsset* mipChainAsset = GetImageMipChainAsset(mipLevel);
+            if (mipChainAsset)
+            {
+                auto mipChainIndex = GetMipChainIndex(mipLevel);
+                auto mipChainOffset = aznumeric_cast<AZ::u32>(GetMipLevel(mipChainIndex));
+                auto layout = mipChainAsset->GetSubImageLayout(mipLevel - mipChainOffset);
+
+                imageDescriptor.m_size = layout.m_size;
+            }
+
+            return imageDescriptor;
+        }
+
         AZStd::span<const uint8_t> StreamingImageAsset::GetSubImageData(uint32_t mip, uint32_t slice)
         {
-            if (mip >= m_mipLevelToChainIndex.size())
+            const ImageMipChainAsset* mipChainAsset = GetImageMipChainAsset(mip);
+
+            if (mipChainAsset == nullptr)
             {
+                AZ_Warning("Streaming Image", false, "MipChain asset wasn't loaded");
```

Comment:
Maybe print the asset ID of the StreamingImageAsset here?

Reply:
Will do!